### PR TITLE
AG-11711 Trim down BeanStub

### PR DESCRIPTION
**Pull request by StephenCooper**
Main changes:

- `context` is now private and removed `getContext`. Those classes that need the context can wire themselves. 
- `gridId` is now on BeanStub as that was the main use case of `getContext`
- `environment ` removed and wired when required

Future work could include removing frameworkOverrides but that might need some more care so leaving out of these changes.

### CODE REVIEW

File: enterprise-modules/master-detail/src/masterDetail/detailCellRenderer.ts [hunk 1]
```diff
@@ -26,10 +28,13 @@ export class DetailCellRenderer extends Component implements ICellRenderer {
...
 
     private detailApi: GridApi;
-
     private params: IDetailCellRendererParams;
-
     private ctrl: DetailCellRendererCtrl;
+    private context: Context;
+
+    public wireBeans(beans: BeanCollection): void {
+        this.context = beans.context;
```
Comment (peterjrreynolds) [inline enterprise-modules/master-detail/src/masterDetail/detailCellRenderer.ts:36]: missing super call

File: community-modules/core/src/gridOptionsService.ts [hunk 2]
```diff
@@ -152,6 +155,78 @@ export class GridOptionsService extends BeanStub {
         return _exists(this.gridOptions[property]);
     }
 
+    private setupGridOptionListener<K extends keyof GridOptions>(
+        event: keyof GridOptions,
+        listener: PropertyValueChangedListener<K>
+    ): () => null {
+        this.gos.addPropertyEventListener(event, listener);
```
Comment (peterjrreynolds) [inline community-modules/core/src/gridOptionsService.ts:162]: Can remove the `.gos`

File: community-modules/core/src/widgets/component.ts [hunk 3]
```diff
@@ -91,7 +91,7 @@ export class Component extends BeanStub implements ComponentBean {
         // ui exists if user sets template in constructor. when this happens,
         // We have to wait for the context to be autoWired first before we can create child components.
         this.agStackComponentsRegistry?.ensureRegistered(this.components);
-        if (element && this.getContext()) {
```
Comment (StephenCooper) [inline community-modules/core/src/widgets/component.ts:94]: Context is not actually being used here, but instead is just a proxy for auto wiring having run. We can tell this from the presence of the existing `agStackComponentsRegistry`

File: enterprise-modules/master-detail/src/masterDetail/detailCellRenderer.ts (hunk 1 above)
Comment (StephenCooper) [reply enterprise-modules/master-detail/src/masterDetail/detailCellRenderer.ts:37]: fixed.

**Pull request merged**